tests/byte_tests.cpp

```diff
@@ -114,6 +114,24 @@ SUITE(byte_tests)
 //      CHECK(0x12 == gsl::to_integer<float>(b));   // expect compile-time error
 //      CHECK(0x12 == gsl::to_integer<double>(b));  // expect compile-time error
     }
+
+    int modify_both(gsl::byte& b, int& i)
+    {
+        i = 10;
+        b = to_byte<5>();
+        return i;
+    }
+
+    TEST(aliasing)
+    {
+        int i{ 0 };
+        int res = modify_both(reinterpret_cast<byte&>(i), i);
+#ifdef GSL_CHECK_BYTE_ALIASING
```

Comment:
I don't think you need to #ifdef out the test. Instead (as per my other comment) add all the flags to avoid the problem with GCC and Clang and then use this test to make sure they work as expected.